modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/dto/v1_0/util/ObjectFieldUtil.java

```diff
@@ -275,6 +275,8 @@ public static com.liferay.object.model.ObjectField toObjectField(
 			}
 		}
 
+		labelMap.putIfAbsent(defaultLocale, objectField.getName());
```

Comment:
Explanation:

From the start (before LPD-22030 was added) the root of the issue was always that some of the locales in the object definition JSON aren't matching the current siteDefaultLanguageId /site default locale.

First, all the locales were of different language than the siteDefaultLanguage. Then the newest issue was that not all the locales are different. Some of them are matching the siteDefaultLanguage, some of them not. For example the actions, and some of the fields were Spanish, but some were English etc.

Solution:
I don't think we should ever compare the locales to the object definition's JSON defaultLanguageId at all. As @marcelabc  stated the defaultLanguageId is not required in the JSON for import, so it could be missing. And as we saw, the locales inside a JSON can be from different languages.
Secondly, the failure was because of the mismatch between the siteDefaultLanguageId and the given locale, not the mismatch between JSON's defaultLanguageId and the locale, so to compare to that is unnecessary.

So my solution is to compare the locales to the current siteDefaultLanguageId, and if there's no locale with the siteDefaultLanguageId in the map, then add to it.

Reply:
@G-Sa could you also check my comments ? :) what do you think?